app/Main.hs

```diff
@@ -96,49 +96,104 @@ writePackageFile =
   . packageConfigToJSON
 
 data PackageInfo = PackageInfo
-  { repo         :: Text
+  { repo         :: Repo
   , version      :: Text
   , dependencies :: [PackageName]
   } deriving (Show, Eq, Generic, Aeson.FromJSON, Aeson.ToJSON)
 
 type PackageSet = Map.Map PackageName PackageInfo
 
+newtype Repo = Repo { unRepo :: Text } deriving (Show, Eq)
+
+instance Aeson.FromJSON Repo where
+  parseJSON = fmap Repo . Aeson.parseJSON
+
+
+instance Aeson.ToJSON Repo where
+  toJSON = Aeson.toJSON . unRepo
+
+
+data CloneTarget = CloneTag Text
+                 | CloneSHA Text
+                 deriving (Show)
+
+toCloneTarget
+  :: Repo
+  -> Text
+  -> IO CloneTarget
+toCloneTarget (Repo from) raw = do
+  remoteLines <- Turtle.fold (lineToText <$> gitProc) Foldl.list
+  let refs = Set.fromList (mapMaybe parseRef remoteLines)
+  if Set.member rawAsBranch refs
+     then do
+       echoT (raw <> " is a branch. psc-package only supports tags and SHAs.")
+       exit (ExitFailure 1)
+     else if Set.member rawAsTag refs
+             then return (CloneTag raw)
+             else return (CloneSHA raw)
+  where
+  rawAsBranch = "refs/heads/" <> raw
+  rawAsTag = "refs/tags/" <> raw
+  gitProc = inproc "git" ["ls-remote", "-q", "--refs", from] empty
```

Comment:
Why `inproc` here? Don't we use `proc` elsewhere? Is there any important difference?